addons/web/static/src/views/fields/state_selection/state_selection_field.js

```diff
@@ -98,7 +98,7 @@ export const stateSelectionField = {
     supportedTypes: ["selection"],
     extractProps({ options, viewType }, dynamicInfo) {
         return {
-            showLabel: viewType === "list" && !options.hide_label,
+            showLabel: viewType.includes('form', 'list') && !options.hide_label,
```

Comment:
I fear that this might impact a lot of existing form views that currently assume that the label is not shown by default.